Code/src/main/java/nl/utwente/group10/ui/components/blocks/Block.java

```diff
@@ -93,8 +97,46 @@ public final CustomUIPane getPane() {
      * change.
      *
      * This method should only be called after the Block's constructor is done.
+     * 
+     * This method will invalidate the Block even if the state did not change.
+     */
+    public void invalidateConnectionState() {
+    }
+
+    /**
+     * Does the same as invalidateConnectionState(), but cascading down to
+     * other blocks which are possibly also (indirectly) affected by the state
+     * change.
+     * 
+     * @param state
+     *            The newest visual state
+     */
+    public void invalidateConnectionStateCascading(int state) {
+        if (!connectionStateIsUpToDate(state)) {
```

Comment:
Heb de aanname erbji gezet.
Over of dit altijd klopt:
Als het betreffende blok up-to-date is, en al zijn kinderen niet, dan moet dit door een andere verandering komen die zijn eigen invalidation triggered. Immers, zo gauw het blok niet up-to-date is worden al zijn kinderen hierover geinformeerd en zijn zij weer up-to-date.